src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.spec.js

```diff
@@ -1,283 +1,271 @@
+  input.focus();
+  expect(input).toHaveFocus();
+  expect(onFocus).toHaveBeenCalled();
+});
+
+it('should call onBlur when input loses focus', () => {
+  const onBlur = jest.fn();
+  const { getByLabelText } = renderLocalizedMultilineTextInput({ onBlur });
+  const input = getByLabelText('EN');
+  input.focus();
+  expect(input).toHaveFocus();
+  input.blur();
+  expect(input).not.toHaveFocus();
+  expect(onBlur).toHaveBeenCalled();
+});
+
+describe('when input is collapsed', () => {
+  it('should allow changing value of the input', () => {
+    const { getByLabelText } = renderLocalizedMultilineTextInput();
+    const event = { target: { value: 'I want chicken' } };
+    const input = getByLabelText('EN');
+    fireEvent.focus(input);
+    fireEvent.change(input, event);
+    fireEvent.keyDown(input, { key: 'Enter' });
+    fireEvent.keyUp(input, { key: 'Enter' });
+    expect(input.value).toEqual('I want chicken');
+  });
+});
+
+describe('when input is expanded', () => {
+  it('should expand and show all language inputs when `Show all languages` is clicked', () => {
+    const { getByLabelText, getByText } = renderLocalizedMultilineTextInput();
+    getByText(/show all languages/i).click();
+    expect(getByLabelText('FR')).toBeInTheDocument();
+  });
+  it('should allow changing the french input', () => {
+    const { getByLabelText, getByText } = renderLocalizedMultilineTextInput();
+    getByText(/show all languages/i).click();
+    const event = { target: { value: 'Je veux manger du poulet' } };
+    const frenchInput = getByLabelText('FR');
+    fireEvent.focus(frenchInput);
+    fireEvent.change(frenchInput, event);
+    fireEvent.keyDown(frenchInput, { key: 'Enter' });
+    fireEvent.keyDown(frenchInput, { key: 'Enter' });
+    expect(frenchInput.value).toEqual('Je veux manger du poulet');
+  });
+});
+
+describe('when expanded by default', () => {
+  it('should render one input per language', () => {
+    const { getByLabelText, getByText } = renderLocalizedMultilineTextInput({
+      areLanguagesDefaultOpened: true,
     });
-    it('should not focus the remaining languages', () => {
-      expect(wrapper.find({ name: 'foo.de' })).toHaveProp(
-        'isAutofocussed',
-        false
-      );
-      expect(wrapper.find({ name: 'foo.es' })).toHaveProp(
-        'isAutofocussed',
-        false
-      );
+    expect(getByLabelText('EN')).toBeInTheDocument();
+    expect(getByLabelText('FR')).toBeInTheDocument();
+    expect(getByText(/hide languages/i)).toBeInTheDocument();
+  });
+});
+
+describe('when expansion controls are hidden', () => {
+  it('should render one input per language and no hide button', () => {
+    const { getByLabelText, queryByText } = renderLocalizedMultilineTextInput({
+      hideLanguageControls: true,
     });
+    expect(getByLabelText('EN')).toBeInTheDocument();
+    expect(getByLabelText('FR')).toBeInTheDocument();
+    expect(queryByText(/hide languages/i)).not.toBeInTheDocument();
```

Comment:
I am just being annoying: is this gonna be a follow up to make these `data-testid` or labels?

Reply:
here [af6476cfbd5fa8212665300a370051a37fce4768](https://github.com/commercetools/ui-kit/pull/314/commits/af6476cfbd5fa8212665300a370051a37fce4768)

Reply:
I think I can could labels here because `AccessibleButton` sets `aria-label` to the label prop.

[ref](https://github.com/kentcdodds/dom-testing-library)